dataproc_jupyter_plugin/services/airflow.py

```
@@ -99,18 +107,16 @@ async def update_job(self, composer_name, dag_id, status):
         airflow_uri, bucket = await self.get_airflow_uri(composer_name)
         try:
             api_endpoint = f"{airflow_uri}/api/v1/dags/{dag_id}"
-            if status == "true":
-                data = {"is_paused": False}
-            else:
-                data = {"is_paused": True}
-            response = requests.patch(
+
+            data = {"is_paused": status.lower() != "true"}
+            async with self.client_session.patch(
                 api_endpoint, json=data, headers=self.create_headers()
-            )
-            if response.status_code == 200:
-                return 0
-            else:
-                self.log.exception("Error updating status")
-                return 1
+            ) as response:
+                if response.status == 200:
+                    return 0
+                else:
+                    self.log.exception("Error updating status")
+                    return 1
```

Return 1?

It should return the structured error response object:

```
return {"error": f"Error updating Airflow DAG status: {response.reason} {await response.text()}"}
```